packages/cli/src/editor/components/RenderModal/RenderModalJSONInputPropsEditor.tsx

```diff
@@ -68,11 +72,17 @@ export const RenderModalJSONInputPropsEditor: React.FC<{
 	onSave,
 	valBeforeSafe,
 	showSaveButton,
+	setIsCustomDateUsed,
 }) => {
 	const keybindings = useKeybinding();
 
 	const [localValue, setLocalValue] = React.useState<State>(() => {
-		return parseJSON(serializeJSONWithDate(value, 2));
+		const serializedJSON: SerializedJSONWithDate = serializeJSONWithDate(
+			value,
+			2
+		);
+		setIsCustomDateUsed(serializedJSON.customDateUsed);
```

Comment:
Not good, it sets a different state while defining a state. Let's refactor it tomorrow so it gets derived 👍